VMEncryption/main/AbstractBekUtilImpl.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+#
+# Azure Disk Encryption For Linux extension
+#
+# Copyright 2016 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import sys
+from Common import TestHooks
+import base64
+import abc
+from abc import abstractmethod
+# compatible with Python 2 *and* 3
+ABC = abc.ABCMeta('ABC', (object,), {'__slots__': ()}) 
+
+class BekMissingException(Exception):
+    """
+    BEK volume missing or not initialized. 
+    add retry-logic to the network api call.
+    """
+    def __init__(self, value):
+       self.value = value
+
+    def __str__(self):
+       return(repr(self.value))
+
+class AbstractBekUtilImpl(ABC):
+    '''
+    This is an interface used for funcitonality implementation for BEK util class
+    '''
+    wrong_fs_msg = "BEK disk does not have vfat filesystem."
```

Comment:
These messages are specific to BEK volume. I believe these should be moved to volume impl class